src/linter/block.go

```diff
@@ -1361,6 +1362,27 @@ func (b *BlockWalker) handleArrayItems(arr node.Node, items []*expr.ArrayItem) b
 			constKey = true
 		case *scalar.Lnumber:
 			key = k.Value
+			constKey = true
+		case *expr.ConstFetch:
+			_, info, ok := solver.GetConstant(b.r.ctx.st, k.Constant)
+
+			if !ok {
+				continue
+			}
+
+			key = info.Value.Value
+
+			if info.Value.Type == meta.Float {
+				val, err := strconv.ParseFloat(key, 64)
+				if err != nil {
+					continue
+				}
+				value := int64(math.Floor(val))
+				key = fmt.Sprint(value)
+			} else if info.Value.Type == meta.Undefined {
```

Comment:
I would suggest moving this check closer to the !ok, so we get guarding clauses above the code that handles correct known values.